cmd/distrogen/registry.go

```diff
@@ -0,0 +1,243 @@
+		tag = "v0.0.0"
+	}
+	return fmt.Sprintf("%s %s", gm.URL, tag)
+}
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+// It takes a properly formed Go Module ID string and unpacks
+// it into the struct.
+func (gm *GoModuleID) UnmarshalYAML(value *yaml.Node) error {
+	// The module ID may have a version.
+	moduleStr := value.Value
+	moduleComponents := strings.Split(moduleStr, " ")
+	gm.URL = moduleComponents[0]
+	if len(moduleComponents) > 1 {
+		gm.Tag = moduleComponents[1]
+	}
+	return nil
+}
+
+// MarshalYAML implements the yaml.Marshaler interface. It leverages
+// the String method to allow outputting the value into a YAML document
+// in the module ID string form.
+func (gm *GoModuleID) MarshalYAML() (interface{}, error) {
+	return gm.String(), nil
+}
+
+// RegistryComponent is the type used as a basis for Registry.
+// It contains all the information needed to output a
+type RegistryComponent struct {
+	GoMod         *GoModuleID `yaml:"gomod"`
+	Import        string      `yaml:"import,omitempty"`
+	Name          string      `yaml:"string,omitempty"`
+	Path          string      `yaml:"path,omitempty"`
+	Stable        bool        `yaml:"stable,omitempty"`
+	StartRevision string      `yaml:"start_revision,omitempty"`
+	DocsURL       string      `yaml:"docs_url"`
+}
+
+// RenderDocsURL renders the docs URL into a template.
+func (c *RegistryComponent) RenderDocsURL() string {
+	if c.DocsURL == "" {
+		return "No docs linked for component"
+	}
+	return c.DocsURL
+}
+
+// IsContrib determines whether the module comes from the opentelemetry-collector-contrib repo.
+func (c *RegistryComponent) IsContrib() bool {
+	return strings.Contains(c.GoMod.URL, "github.com/open-telemetry/opentelemetry-collector-contrib")
+}
+
+// OCBManifestComponent is a reflection of the fields for an
+// entry in an OCB manifest yaml.
+type OCBManifestComponent struct {
+	GoMod  *GoModuleID `yaml:"gomod"`
+	Import string      `yaml:"import,omitempty"`
+	Name   string      `yaml:"string,omitempty"`
+	Path   string      `yaml:"path,omitempty"`
+}
+
+// GetOCBComponent will return an OCBManifestComponent using
+// the details from this RegistryComponent.
+func (c *RegistryComponent) GetOCBComponent() OCBManifestComponent {
+	return OCBManifestComponent{
+		GoMod:  c.GoMod,
+		Import: c.Import,
+		Name:   c.Name,
+		Path:   c.Path,
+	}
+}
+
+// RegistryComponents is a map of registry component names to component
+// details.
+type RegistryComponents map[string]*RegistryComponent
+
+// LoadAllComponents will take a list of component names and load them
+// from the registry, attaching the appropriate version tag.
+func (rl RegistryComponents) LoadAllComponents(names []string, version string, stableVersion string, contribVersion string) (RegistryComponents, RegistryLoadError) {
+	components := RegistryComponents{}
+	errs := make(RegistryLoadError)
+
+	for _, name := range names {
+		entry, err := rl.LoadComponent(name)
```

Comment:
[nit] This is just a map lookup now, we can inline this when its called and get rid of this method.

```suggestion
		entry, ok := rl[name]
        if !ok {
            errs[name] = ErrComponentNotFound
            continue
        }
```

Reply:
I ended up refactoring this slightly in [055331f](https://github.com/GoogleCloudPlatform/opentelemetry-operations-collector/pull/246/commits/055331f9c511d432aa04db531625fe518ebb0de8)

The method is still there but it does things in a different order that makes more sense